sale_advance_payment/models/account_move.py

```diff
@@ -9,18 +9,31 @@ class AccountMove(models.Model):
 
     def _post(self, soft=True):
         # Automatic reconciliation of payment when invoice confirmed.
-        res = super(AccountMove, self)._post(soft=soft)
+        res = super()._post(soft=soft)
         for move in self:
             sale_order = move.mapped("line_ids.sale_line_ids.order_id")
-            if (
-                not sale_order
-                or move.invoice_outstanding_credits_debits_widget is False
-            ):
+            if not sale_order:
                 continue
-            json_invoice_outstanding_data = (
-                move.invoice_outstanding_credits_debits_widget.get("content", [])
+
+            # Get payment move IDs first
+            payment_move_ids = sale_order.account_payment_ids.move_id.ids
+            if not payment_move_ids:
+                continue
+
+            # Domain search (most efficient for large datasets)
+            advance_payment_lines = self.env["account.move.line"].search(
```

Comment:
@ljsalvatierra-factorlibre For a better performance, do the search before the loop, then, filter records on payment_move_ids.

Reply:
Yes!